signer-wasm/src/ledger.rs

```diff
@@ -287,25 +223,10 @@ pub async fn device_info(transport_wrapper: TransportWrapper) -> Promise {
     let apdu_transport = APDUTransport {
         transport_wrapper: tmp,
     };
-
     let app = filecoin_signer_ledger::app::FilecoinApp::new(apdu_transport);
 
-    let d_result = app.get_device_info().await;
-
-    match d_result {
-        Ok(d) => Promise::resolve(&ok_or_ret_promise!(
-            JsValue::from_serde(&d),
-            "Error converting error message to javascript value."
-        )),
-        Err(err) => {
-            let error = Error {
-                return_code: 0x6f00,
-                error_message: err.to_string(),
-            };
-            Promise::reject(&ok_or_ret_promise!(
-                JsValue::from_serde(&error),
-                "Error converting error message to javascript value."
-            ))
-        }
+    match app.get_device_info().await {
+        Ok(d) => Promise::resolve(&js_or_reject!(&d)),
+        Err(err) => Promise::reject(&js_or_reject!(&Error::new(0x6f00, err.to_string()))),
```

Comment:
The tests didn't fail but I think this logic is wrong because if the encoding fails, then a promise is returned and it is like returning a promise that returns a promise.

```suggestion
        Err(err) => {
            let encoded = js_or_reject!(&Error::new(0x6f00, err.to_string()));
            Promise::reject(&encoded)
        }
```